src/hotspot/share/opto/split_if.cpp

```diff
@@ -83,8 +85,231 @@ bool PhaseIdealLoop::split_up( Node *n, Node *blk1, Node *blk2 ) {
     }
   }
 
+  if (process_cmp_loadklass(n, blk1, blk2)) {
```

Comment:
I suggest to also rename this into something like `clone_cmp_loadklass_down` to follow the naming of `clone_cmp_down`.